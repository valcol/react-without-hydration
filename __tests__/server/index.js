/**
 * @jest-environment node
 */

import React from "react";
import ReactDOMServer from "react-dom/server";
import withoutHydrate from "../../src";

const Component = ({ label }) => <div className="label">{label}</div>;

test("Render correctly server side ", () => {
    const ComponentWithoutHydrateSRR = withoutHydrate()(Component);

    const component = ReactDOMServer.renderToString(
        <ComponentWithoutHydrateSRR label="some content server side" />
    );
    expect(component).toMatchSnapshot();
});
