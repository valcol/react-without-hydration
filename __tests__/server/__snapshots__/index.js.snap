// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly server side  1`] = `"<section data-no-hydrate=\\"true\\" data-reactroot=\\"\\"><div class=\\"label\\">some content server side</div></section>"`;
